lib/hmmlearn/tests/test_base.py

```diff
@@ -149,6 +149,22 @@ def test_score_samples(self):
                                   [0.8673, 0.1327]])
         assert np.allclose(posteriors, refposteriors, atol=1e-4)
 
+    def test_score_batches_log(self):
+        # ``StubHMM` ignores the values in ```X``, so we just pass in an
+        # array of the appropriate shape.
+        logprobs = self.hmm._score_batches_log(self.framelogprob)
+
+        reflogprob = -3.3725
+        assert round(sum(logprobs), 4) == reflogprob
```

Comment:
perhaps make a more stringent test?

Reply:
Even just a single function that loops over the various `implementations` - that is how I introduced the log/scaling logic.